### PR TITLE
Add dynamic query, configurable cursor module

### PR DESCRIPTION
**Pull request by jeroenvisser101**
This removes the limit to 3 sorts, and allows custom cursor translator
implementations.

I also altered the default cursor spec to be 'tamper resistant', basically avoiding errors when the cursor is altered (such as the last character being dropped). Let me know if this is OK, otherwise we can ship two cursor translators, tamper-resistant and basic?

### CODE REVIEW

File: test/support/priv/repo/migrations/20210924142814_create_users.exs [hunk 1]
```diff
@@ -5,6 +5,7 @@ defmodule AbsintheRelayKeysetConnection.Repo.Migrations.CreateUsers do
     create table(:users) do
       add(:first_name, :string)
       add(:last_name, :string)
+      add(:inserted_at, :naive_datetime_usec)
```
Comment (jeroenvisser101) [inline test/support/priv/repo/migrations/20210924142814_create_users.exs:8]: I'm still working on tests for this one. Works great, just want to incorporate tests on a non-string column :)

**Comment by nathanl** [comment]
Hey @jeroenvisser101, thanks for the PR! Would you mind breaking it up into smaller ones, though? The use of `dynamic` for the `WHERE` clauses one be one, the cursor change another, and the `naive_datetime` another.

**Comment by nathanl** [comment]
I love the reduction in code and increase in flexibility with `dynamic`. 😄

**Comment by jeroenvisser101** [comment]
> Hey @jeroenvisser101, thanks for the PR! Would you mind breaking it up into smaller ones, though? The use of `dynamic` for the `WHERE` clauses one be one, the cursor change another, and the `naive_datetime` another.

For sure! There's also formatting changes (mainly ensuring all typespecs have parens, so remote, kernel and local types all use parens), shall I send this one as the first PR, then base the others on top of it? 

> I love the reduction in code and increase in flexibility with `dynamic`. 😄

Yeah, I was surprised how well it worked. It took me a couple of tries, but working from "outside to inside" in the clauses worked very well and kept the clauses grouped with parens, and allowed to reduce the number of total conditions in the query (like the original did).

**Comment by nathanl** [comment]
> shall I send this one as the first PR, then base the others on top of it?

👍

**Comment by jeroenvisser101** [comment]
Just now had time for this, hope you're able to review them, let me know if any other changes are required 👍

**Comment by nathanl** [comment]
Thank you! I can't review yet but hopefully next week.

**Pull request closed without merging**